### ISSUE TITLE
Chore sensor does not contain user assignments

### ISSUE THREAD

**Issue by zigurana** [custom-components/grocy#44]
**Describe the bug**
The Grocy API supplies `next_execution_assigned_to_user_id`, but this is not reported by the sensor.
Therefore it cannot be used as an info source for Lovelace Cards etc.
(Grocy Custom Component v. 0.4.0)

**To Reproduce**
1. Go to 'HomeAssistant > Development tools > States'
2. Select sensor.grocy_chores entity
3. Inspect values for chores, see:
```
  - _chore_id: 1
    _last_tracked_time: '2020-04-07T00:00:00+00:00'
    _next_estimated_execution_time: '2020-04-12T00:00:00+00:00'
    _name: Planten watergeven
    _last_done_by:
      _id: 3
      _username: Jannica
      _first_name: Janneke
      _last_name: ''
      _display_name: Janneke
  - _chore_id: 2
    _last_tracked_time: '2020-04-08T00:00:00+00:00'
    _next_estimated_execution_time: '2020-04-09T00:00:00+00:00'
    _name: Afwassen
    _last_done_by:
      _id: 1
      _username: admin
      _first_name: null
      _last_name: null
      _display_name: admin
```
4. No entry for `_next_execution_assigned_to_user_id` is found.

**Expected behavior**
`next_execution_assigned_to_user_id` should also be part of the sensor data.

**Are you using HASSIO to run grocy?**
Yes

**configuration.yaml entry**
```yaml
default
```
Grocy config = default

**JSON service data (if related to using a service)**
```json
{

}
```

**Additional context**

**Comment by SebRut** [comment]
You're right! This means that we have to make a separate call for every chore as it isn't included in the default fetch-all-chores call. Implementation tracked [here](https://github.com/SebRut/pygrocy/issues/87).

**Comment by zigurana** [comment]
That's the thing, when I run `URL/api/chores` (which is the fetch-all-chores you mentioned, I think), I recieve:
```
{
    "chore_id": "1",
    "last_tracked_time": "2020-04-07 00:00:00",
    "next_estimated_execution_time": "2020-04-14 00:00:00",
    "track_date_only": "1",
    "next_execution_assigned_to_user_id": null
  },
```
What I do not understand, is that the actual name of the chore is not even part of that listing! So are you currently already iterating over all valid chore ideas to capture their details (using `/api/chores/[choreId]`)?
In any case, the `next_execution_assigned_to_user_id` attribute is included in both calls.

So at least the user id is reported. From there it would be just one other call to `GET URL/api/users` to retrieve the display names for those id's, correct?

**Comment by SebRut** [comment]
I'm basing my assumption on the API spec of the grocy demo instance (https://en.demo.grocy.info/api#/Chores/get_chores__choreId_) which doesn't include `next_execution_assigned_to_user_id`. It shouldn't be hard to implement either way, I just have to find some time to do it.

**Comment by zigurana** [comment]
If you want to give some pointers to get me started, I'd be happy to take a look.

**Comment by KibosJ** [comment]
> I'm basing my assumption on the API spec of the grocy demo instance (https://en.demo.grocy.info/api#/Chores/get_chores__choreId_) which doesn't include `next_execution_assigned_to_user_id`. It shouldn't be hard to implement either way, I just have to find some time to do it.

You shouldn't have to look up the user ID for this as the response includes that information.

The full output of my api call to /api/chores/1 for instance is

`{
    "chore": {
        "id": "1",
        "name": "Wash dishes",
        "description": "",
        "period_type": "daily",
        "period_days": "0",
        "row_created_timestamp": "2020-05-30 07:32:32",
        "period_config": "",
        "track_date_only": "1",
        "rollover": "1",
        "assignment_type": "random",
        "assignment_config": "1",
        "next_execution_assigned_to_user_id": "1",
        "consume_product_on_execution": "0",
        "product_id": "",
        "product_amount": null,
        "period_interval": "1"
    },
    "last_tracked": "2020-05-29 00:00:00",
    "tracked_count": 1,
    "last_done_by": {
        "id": "1",
        "username": "James",
        "first_name": "James",
        "last_name": "",
        "row_created_timestamp": "2020-04-17 13:19:17",
        "display_name": "James"
    },
    "next_estimated_execution_time": "2020-05-30 00:00:00",
    "next_execution_assigned_user": {
        "id": "1",
        "username": "James",
        "first_name": "James",
        "last_name": "",
        "row_created_timestamp": "2020-04-17 13:19:17",
        "display_name": "James"
    }
}`

So you could just reference the "next_execution_assigned_user" right?

**Comment by isabellaalstrom** [comment]
@SebRut I can see that you updated pygrocy for this. Can I help with something on the component sida for this?

**Comment by SebRut** [comment]
@isabellaalstrom Thank you for the offer! It seems like my grocy instance stopped working recently, so I can't test stuff atm. I guess you could install the newest component version from `develop` branch and check if the component is working. The chore objects should probably contain the new data, but if I remember correctly something wasn't working in regard to serialization to json.

**Comment by isabellaalstrom** [comment]
Yes, I actually did that yesterday and it filled up the logs with something with json serialization not working, to the point that I had to stop home assistant from docker and remove the component just to be able to downgrade it again 😄 I can see if I can get it up on my dev env instead so I can see more what's causing that.

**Comment by SebRut** [comment]
Sounds like what I experienced :D I don't really see while the components internal serialization should stop working so I guess it's related to serialization in something like the HAs history/logbook component.

**Comment by Gibby** [comment]
Following. Let me know if you need a tester.

**Comment by SebRut** [comment]
_Hey! Just wanted to let you know that I finally accepted, that I'll not be able to maintain this project for the foreseeable future. I'd be happy to hand most of this over to someone willing to work on this integration and will try to support this by working on SebRut/pygrocy._

**Comment by Gibby** [comment]
I do not consider myself a developer at all but I can usually read through code and come up with small fixes. 

I'll have a PR for this, if the PR I created at https://github.com/SebRut/pygrocy/pull/106 gets accepted. It is based of off pygrocy v12.2 and not the latest which is v15.0. My hope is that it is accepted and a release of v12.3 will be created from it.

v15.0 just had too many changes from v12.2 that I was able to get working.

If you want to test it out:

- Install v0.4.2 of this repo from https://github.com/Gibby/grocy/tree/v0.4.2
- Try out the grocy-chores-card.js from https://github.com/Gibby/lovelace-grocy-chores-card/tree/next_user_id

**Comment by isabellaalstrom** [comment]
> _Hey! Just wanted to let you know that I finally accepted, that I'll not be able to maintain this project for the foreseeable future. I'd be happy to hand most of this over to someone willing to work on this integration and will try to support this by working on SebRut/pygrocy._

How does it work if someone wants to take over maintaining it? Does it have to be here in the "custom components" repo, I don't now how this works?

**Comment by SebRut** [comment]
@isabellaalstrom I guess you'd have to contact an admin of the custom-component organization, maybe @ludeeus ?

**Comment by isabellaalstrom** [comment]
I think I already am a part of the organization actually.. 🤔

**Comment by SebRut** [comment]
Hmm... @iantrich I think you added me to the repository originally, can you help here?

**Comment by ludeeus** [comment]
@isabellaalstrom now have the same level of permissions as @SebRut (write), should one of you be admin maybe?, then you can handle this yourself

**Comment by isabellaalstrom** [comment]
Yeah, I can be admin if that helps.
@SebRut How does the pygrocy repo feel? Do you update it or is it stale atm?

**Comment by SebRut** [comment]
@isabellaalstrom I try to atleast maintain the repo in the sense of reviewing and merging PRs and stuff. Not sure what amount of development will be done by me.

**Comment by isabellaalstrom** [comment]
Okey, now me and @ludeeus have fixed this. A pre-release of 0.5.0 is available for testing, and the chores sensor now contains more attributes for each chore. Try it out and get back to me!
There is also a pre-release for the card which contains fixes to get it working with 0.5.0 of the component. No new functionality yet.
